### PR TITLE
Documentation for report generation

### PR DESCRIPTION
**Pull request by rozsatib**
Documentation and examples for graphgen.py and htmlgen.py usage.

Small fixes for readability and usability to both scripts.

### CODE REVIEW

**Comment by wentasah** [comment]
Can you rebase your branch on top of master, to get rid of the first commit, which is already in the upstream repository, but modified?

File: report_gen/README.md [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+
+### Basic functionality
+
+Read supplied csv files, create graphs from the columns they have in common, and store them. As default, the x axis is the first column in the first file.  
+
+Ignores the first line of the csv file, as it commonly holds the name of the command which generated it.
+
+Output filenames are of the format: all:x_axis-y_axis.extension
+
+####CSV files used in example:
```
Comment (joelmatejka) [inline report_gen/README.md]:
```suggestion
#### CSV files used in example:
```

File: report_gen/README.md [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+    x0, y1, y2, y3         y1, x0, y2, z4         x0, y1, y3, z4
+    1,  2,  8,  4          1,  1,  9,  5          1,  4,  8,  6
+    2,  4,  4,  3          2,  3,  5,  4          2,  6,  4,  5
+    3,  8,  2,  2          4,  7,  3,  3          3,  7,  2,  4
+
+As seen from the structure of the csv files:  
+1. The order of columns can be different from file to file  
+2. Not all files have to contain the same columns
+
+#### Example usage:
```
Comment (joelmatejka) [inline report_gen/README.md]: I would prefer the following structure:
## Using graph generator (graphgen.py)
### Arguments
### Basic functionality
### Input files specification
**Input files example**
### Examples of usage
#### Basic usage
**Sample command**
**Sample output**
#### Filter columns
...
#### Custom x axes
...
#### Grouping data from multiple files
#### Output files extension

File: report_gen/README.md [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+<img src="./example/figs/graphgen/all:x0-y1.svg" width="300" height="auto"> | <img src="./example/figs/graphgen/all:x0-y2.svg" width="300" height="auto">
+
+|example/figs/graphgen/all:x0-y1.svg|example/figs/graphgen/all:x0-y2.svg
+:-------------------------:|:-------------------------:
+<img src="./example/figs/graphgen/all:x0-y3.svg" width="300" height="auto"> | <img src="./example/figs/graphgen/all:x0-z4.svg" width="300" height="auto">
+
+### Features
+#### 1. Plot by category
+
+Plots graphs by category: for each test separately, parses columns for search strings and plots them with a common name on the y axis. The search string and the name are input arguments in dict form: '{"searchstring":"name"}'.
```
Comment (joelmatejka) [inline report_gen/README.md]: For me, it was quite difficult to understand what is the category and what is "only plot specific columns". If I understand correctly. (1) is more like filtering columns with some pattern and (3) sounds to me like grouping the same columns from multiple files to one graph.

File: report_gen/README.md [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+:-------------------------:|:-------------------------:|:-------------------------:
+<img src="./example/figs/graphgen/a:x0-Ytitle.svg" width="300" height="auto"> | <img src="./example/figs/graphgen/b:x0-Ytitle.svg" width="300" height="auto"> | <img src="./example/figs/graphgen/c:x0-Ytitle.svg" width="300" height="auto">
+
+|example/figs/graphgen/b:x0-Ztitle.svg|example/figs/graphgen/c:x0-Ztitle.svg|
+:-------------------------:|:-------------------------:
+<img src="./example/figs/graphgen/b:x0-Ztitle.svg" width="300" height="auto"> | <img src="./example/figs/graphgen/c:x0-Ztitle.svg" width="300" height="auto">
+
+#### 2. Plot on multiple custom x axes
+
+The input x axes do not have to match with the columns perfectly; e.g. -x y would plot with x axes y1,y2,y3.
```
Comment (joelmatejka) [inline report_gen/README.md]: I did not get what it does from this sentence. I would call that "custom x axes" and say that with this argument you can select source data for x axes...

File: report_gen/README.md [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+##### Example:
+    python graphgen.py -f example/a.csv example/b.csv example/c.csv -o example/figs/graphgen --plot_col y1 -e png svg html
+
+##### Generated graphs: 
+|example/figs/graphgen/all:x0-y1.png|example/figs/graphgen/all:x0-y1.svg|
+:-------------------------:|:-------------------------:
+<img src="./example/figs/graphgen/all:x0-y1.png" width="300" height="auto"> | <img src="./example/figs/graphgen/all:x0-y1.svg" width="300" height="auto">
+
+**example/figs/graphgen/all:x0-y1.html**
+<script src=\"https://cdn.plot.ly/plotly-latest.min.js\"></script><iframe frameborder=0 width="600" height="400" src=./example/figs/graphgen/all:x0-y1.html></iframe>
```
Comment (joelmatejka) [inline report_gen/README.md]: This  <script> line is ugly and does not work on github, just write see example in example/... or create screenshot instead

File: report_gen/README.md [hunk 6]
```diff
@@ -0,0 +1,157 @@
...
+
+
+## Using htmlgen.py
+
+### Arguments: 
+    -f,--fig_dir - Path to the figures directory ( Required )
+    -t, --thumbnail_type - Extension of thumbnails on the main page ( Required )
```
Comment (joelmatejka) [inline report_gen/README.md]: If you select thumbnail type to html, it is not readable and clickable at all. Solution might be to set min width/height of the table cell or and add text link below the figure in the case of html thumbnails.

File: report_gen/README.md [hunk 7]
```diff
@@ -0,0 +1,157 @@
...
+    -g, --graph_type - Extension of graphs ( Required )
+    -o, --out_dir - Output directory for the report ( Default: current dir )
```
Comment (joelmatejka) [inline report_gen/README.md]: In both scripts, please create the directory automatically. Now you have to create directories manually first, which is a bit annoying.

File: report_gen/example/a.csv [hunk 8]
```diff
@@ -0,0 +1,5 @@
+Generated by: ./thiscommand
```
Comment (joelmatejka) [inline report_gen/example/a.csv]: Thermobench creates first line with hash at the beginning...

File: report_gen/README.md (hunk 3 above)
Comment (rozsatib) [reply report_gen/README.md]: (1) is filtering columns with a specific pattern in each file and plotting them into a single graph. So for 3 files, 3 graphs are generated for each category.

As default, columns with the same name are plotted to a graph, for all columns found in the files. So for N files and 4 unique column names we get 4 graphs.
(3) limits these columns to those which contain some pattern. So from the 4 column names, only the ones containing "string" have a graph generated.

I will try to improve the descriptions of (1) and (3), but I am unsure how to best explain this.

**Comment by rozsatib** [comment]
All suggested changes resolved/implemented.

**Comment by wentasah** [comment]
I'm closing this pull request and opening a new one. This one is somewhat broken, and new commits cannot be seen and reviewed here. This is even documented at https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/about-pull-requests:

> When pushing commits to a pull request, don't force push. Force pushing can corrupt your pull request.

**Pull request closed without merging**